coordinator/impl/shard_controller.go

```diff
@@ -368,6 +368,10 @@ func (s *shardController) electLeader() error {
 
 func (s *shardController) deletingRemovedNodes() error {
 	for _, ds := range s.shardMetadata.RemovedNodes {
+		// todo: merge to single request
+		if _, err := s.newTerm(s.ctx, ds); err != nil {
```

Comment:
Why do we need to trigger a new term when removing nodes?

Reply:
We have a check on the server side. The follower will reject the deleting request if the term does not fit.